_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+has its own *frequency*,
+is *someone's responsibility*,
+and generates *specific products*.
+If you have a description like this of how your team operates,
+please [get in touch](mailto:{{site.author.email}}):
+I think we could learn a lot from comparisons.
+
+## Company Goals and Priority Projects
+
+This activity happens up in the stratosphere.
+As an engineering manager I may occasionally be called on for input,
+but for the most part this is the ground for everything else described below.
+
+-   Summary: Prioritized company objectives and initiatives for the year
+-   Key Questions:
+    -   What should we invest in/focus on?
+-   Frequency: 6-24 months
+-   Responsible: executive team 
+-   Contributors: company leadership and technical experts at all levels
+-   Products:
+    -   Monthly business review
+    -   Annual all-hands meeting
+
+## Engineering Team
+
+Our team includes about a dozen people
+and is responsible for half that number of applications,
+ranging from complex C++ libraries wrapped up in Python
+to three-tier web applications for managing experimental data
+and command-line tools to support routine tasks.
+The activities below are done by the whole team.
+
+### Engineering OKRs
+
+The VP of Engineering defines objectives and key results at the start of each quarter
+based on discussions with senior management,
+the leaders of other teams,
+and members of our team.
+Goals are prioritized as P0 (not doing this will block something critical to the company),
+P1 (we need this, but we can scale it back or it can slip if a P0 overruns),
+and P2 (it would be nice to have).
+
+-   Summary: Engineering-specific objectives and projects based on:
+    1.  company goals
+    2.  users' needs
+    3.  technical opportunities
+-   Key Questions:
+    -   What should each engineering team focus on and why?
+-   Frequency: quarterly
+-   Responsible: VP Engineering
+-   Contributors: entire engineering team
+-   Products:
+    -   Prioritized list of objectives and key results
+
+> Note: we used to define more of these each quarter than we do now.
+> The theory was that if we're meeting all of our targets,
+> our targets weren't ambitious enough.
+> We now realize that setting ourselves up for continual failure like that isn't actually helpful.
+
+### Engineering OKR Reviews
+
+Improvement in anything requires [reflective practice][reflection].
+Our end-of-quarter OKR reviews therefore have two purposes:
+to make sure everyone understands where we actually are,
+and to summarize what we've learned so that we can do better next time.
+
+-   Summary: End of quarter sync on OKR progress
+-   Key Questions:
+    -   What is the status of each OKR?
+    -   What progress did each team make towards overall goals?
+    -   What losses, challenges, and obstacles did we encounter?
+    -   What did we learn?
+-   Frequency: quarterly
+-   Responsible: VP Engineering
+-   Contributors: project leads and other team members
+-   Products:
```

Comment:
Suggestion: Another product here is a final progress report on each of our OKRs? You allude to that in your summary, but it could be useful to call out here as well

Reply:
amended